cpu/cc26x0/periph/timer.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2016 Leon George
+ * (who won't go after you should you copy any part of this file ;-)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+/**
+ * @ingroup     driver_periph
+ * @{
+ *
+ * @file
+ * @brief       low-level timer driver implementation for the CC26x0 CPU
+ *
+ * @author      Leon M. George <leon@georgemail.eu>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "board.h"
+#include "cpu.h"
+#include "sched.h"
+#include "thread.h"
+#include "periph/timer.h"
+#include "periph_conf.h"
+
+#include "cc26x0-prcm.h"
+
+typedef struct {
+    void (*cb)(int);
+} timer_info_t;
+
+timer_info_t config[TIMER_NUMOF];
+#define USECS_PER_SEC        (1000 * 1000)
+
+static GPT_REG_t *select_gpt(tim_t dev)
+{
+    return timer_conf[dev].gpt;
+}
+
+int timer_init(tim_t dev, unsigned int ticks_per_us, void (*callback)(int))
+{
+    GPT_REG_t *gpt = select_gpt(dev);
+
+    unsigned int gpt_num = ((uintptr_t) gpt - (uintptr_t) GPT0_BASE) / 0x1000;
+
+    config[dev].cb = callback;
+
+    PRCM->GPTCLKGR |= (1 << gpt_num);
+    PRCM->GPTCLKGS |= (1 << gpt_num);
+    PRCM->CLKLOADCTL = CLKLOADCTL_LOAD;
+    while (!(PRCM->CLKLOADCTL & CLKLOADCTL_LOADDONE)) ;
+
+    gpt->CTL = 0;
+
+    gpt->CFG  = GPT_CFG_32T;
+    gpt->TAMR = GPT_TXMR_TXMR_PERIODIC | GPT_TXMR_TXCDIR_UP;
+
+    gpt->TAPR = RCOSC48M_FREQ / (ticks_per_us * USECS_PER_SEC) - 1;
+
+    timer_irq_enable(dev);
+
+    gpt->CTL |= GPT_CTL_TAEN;
+
+    return 0;
+}
+
+int timer_set(tim_t dev, int channel, unsigned int timeout)
+{
+    return timer_set_absolute(dev, channel, timer_read(dev) + timeout);
+}
+
+int timer_set_absolute(tim_t dev, int channel, unsigned int value)
+{
+    if (channel != 0)
+        return -1;
+
+    timer_conf[dev].gpt->TAILR = value;
+    timer_conf[dev].gpt->IMR |= GPT_IMR_TATOIM;
+
+    return 1;
+}
+
+int timer_clear(tim_t dev, int channel)
+{
+    timer_conf[dev].gpt->CTL &= (~GPT_CTL_TAEN);
+
+    return 1;
+}
+
+unsigned int timer_read(tim_t dev)
+{
+    return timer_conf[dev].gpt->TAR;
+}
+
+void timer_stop(tim_t dev)
+{
+    timer_conf[dev].gpt->CTL &= (~GPT_CTL_TAEN);
+}
+
+void timer_start(tim_t dev)
+{
+    timer_conf[dev].gpt->CTL |= GPT_CTL_TAEN;
+}
+
+void timer_irq_enable(tim_t dev)
+{
+    NVIC_EnableIRQ(timer_conf[dev].irqn);
+}
+
+void timer_irq_disable(tim_t dev)
+{
+    NVIC_DisableIRQ(timer_conf[dev].irqn);
+}
+
+static void _interrupt(tim_t dev, int chan)
+{
+    uint32_t mask = GPT_IMR_TATOIM;
+
+    /* might want to remove this */
+    if (timer_conf[dev].gpt->MIS & (~timer_conf[dev].gpt->IMR))
+    {
+        printf("unexpected interrupt (%lx)\n", timer_conf[dev].gpt->MIS);
```

Comment:
don't use printf for error reporting, either use `log` of just don't report this (as this case above should be never happening anyway if the rest of the timer code is ok...)